.github/workflows/static-analysis-and-test.yml

```diff
@@ -121,10 +121,18 @@ jobs:
         run: |
           tox -e end
 
-      # TODO: Figure out how to enable coverage reporting when running on forks
-      # For now the previous step at least puts the coverage reporting in the
-      # github action log.
-      # - name: Pytest Coverage Comment
-      #   uses: MishaKav/pytest-coverage-comment@main
-      #   with:
-      #     pytest-xml-coverage-path: ./coverage.xml
+          # Report and write to summary.
+          python -m coverage report --format=markdown >> $GITHUB_STEP_SUMMARY
+
+          # Write html coverage report to upload as an artifact
+          python -m coverage html
+
+          # Report again and fail if under 100%.
+          python -m coverage report --fail-under=100
+
+      - name: Upload HTML report if check failed.
+        uses: actions/upload-artifact@v3
+        with:
+          name: html-report
+          path: htmlcov
+        if: ${{ failure() }}
```

Comment:
Btw, we are going to be commenting out the `--fail-under`  and `failure()` lines for now. I would love to have 100% test coverage but we have to figure out how to do Qt gui testing before we can expect test coverage.

See how I disabled it in [hab](https://github.com/blurstudio/hab/commit/6af8313a8d0e037693a94ca93b5a9f8f2884a4cc). I want this to be its own commit in the history, so we can add it after squashing.

Looks like the actions are passing other than this check, so once that change is made all tests are passing, though with 0% coverage but we will need to handle that in future PR's.

Reply:
Take another look at the hab commit I linked. You missed commenting out the `--fail-under` line. That line is what actually makes the coverage job fail. The changes you applied just makes it so we always upload the html-report artifact, instead of just uploading it when we don't have 100% coverage.